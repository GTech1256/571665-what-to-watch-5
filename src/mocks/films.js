import {getGenresByFilms} from "../bl/film";
import reviews from "./reviews";

export const films = [
  {
    id: 1,
    name: `The Grand Budapest Hotel`,
    posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    backgroundImage: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://some-link`,
    previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    rating: 8.9,
    scoresCount: 240,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
    runTime: 99,
    genre: `Comedy`,
    released: 2014,
    isFavorite: false,
    reviews: reviews.slice(0, 2)
  },
  {
    id: 2,
    name: `Fantastic Beasts`,
    posterImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://some-link`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `Fantastic Beasts purports to be a reproduction of a textbook owned by Harry Potter and written by magizoologist Newt Scamander, a character in the fictional Harry Potter series. In the series, magizoology is the study of magical creatures`,
    rating: 7.3,
    scoresCount: 338,
    director: `David Yates`,
    starring: [`Eddie Redmayne`, `Katherine Waterston`, `Alison Sudol`],
    runTime: 132,
    genre: `Fantasy`,
    released: 2016,
    isFavorite: true,
    reviews: reviews.slice(2)
  },
];

export const genres = getGenresByFilms(films);
