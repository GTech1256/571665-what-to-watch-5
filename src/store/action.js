export const ActionType = {
  TOGGLE_GENRE_FILTER: `TOGGLE_GENRE_FILTER`,
};

export const ActionCreator = {
  toggleGenreFilter: (genre) => ({
    type: ActionType.TOGGLE_GENRE_FILTER,
    payload: genre
  }),
};
