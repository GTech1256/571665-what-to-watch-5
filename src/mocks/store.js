
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {NameSpace} from "../store/reducers/root-reducer";
import film from './film';
import films from './films';
import review from './review';

const FILMS_SHOWED_PER_STEP_COUNT = 8;
const FIRST_FILTER_NAME = `All Genres`;
const EMPTY_STATE_VALUE = null;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const mockStore = configureStore([thunk]);

export default mockStore({
  [NameSpace.ADJUSTMENT]: {
    showedFilmsCount: FILMS_SHOWED_PER_STEP_COUNT,
    activeFilterGenre: FIRST_FILTER_NAME,
  },
  [NameSpace.DATA]: {
    promoFilm: film,
    film,
    films,
    filters: [FIRST_FILTER_NAME, `Action`, `Comedy`],
    comments: [review],
    favoriteFilms: films,
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.NO_AUTH,
    user: EMPTY_STATE_VALUE
  },
});
