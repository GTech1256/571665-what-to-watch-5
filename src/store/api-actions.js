import {loadFilms, loadPromoFilm, setGenres} from "./action";
import {APIRoute} from "../const";
import {getGenresByFilms} from "../bl/film";
import {adaptFilmFromServer} from "../utils/filmAdapter";

export const fetchPromoFilm = () => (dispatch, _getState, api) => (
  api.get(APIRoute.PROMO_FILM)
    .then(({data}) => dispatch(loadPromoFilm(adaptFilmFromServer(data))))
);

export const fetchFilmsList = () => (dispatch, _getState, api) => (
  api.get(APIRoute.FILMS)
    .then(({data}) => {
      const films = data.map(adaptFilmFromServer);

      dispatch(loadFilms(films));
      dispatch(setGenres(getGenresByFilms(films)));
    })
);

// export const checkAuth = () => (dispatch, _getState, api) => (
//   api.get(APIRoute.LOGIN)
//     .then(() => dispatch(requireAuthorization(AuthorizationStatus.AUTH)))
//     .catch((err) => {
//       throw err;
//     })
// );

// export const login = ({login: email, password}) => (dispatch, _getState, api) => (
//   api.post(APIRoute.LOGIN, {email, password})
//     .then(() => dispatch(requireAuthorization(AuthorizationStatus.AUTH)))
//     .then(() => dispatch(redirectToRoute(AppRoute.RESULT)))
// );
