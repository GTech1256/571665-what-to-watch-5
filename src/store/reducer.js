import {ActionType} from "./action";
import {films, genres} from "../mocks/films";
import {extend} from "../utils/extend";
import {FILM_ALL_GENRES} from "../const";
import {getFilmsByGenre} from "../bl/film";

const initialState = {
  promoFilm: films[0],
  allFilms: films,
  films,
  genre: FILM_ALL_GENRES,
  genres
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.TOGGLE_GENRE_FILTER:
      const genre = action.payload;

      return extend(state, {
        genre,
        films: getFilmsByGenre(initialState.films, genre)
      });

    default:
      return state;
  }
};

export {reducer};
