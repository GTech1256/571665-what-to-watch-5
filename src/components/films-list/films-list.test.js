import React from "react";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import films from "../../mocks/films";
import store from "../../mocks/store";
import FilmsList from "./films-list";

it(`FilmsList is rendered correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <Provider store={store}>
            <FilmsList
              limit={2}
              films={films}
            />
          </Provider>
        </MemoryRouter>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
