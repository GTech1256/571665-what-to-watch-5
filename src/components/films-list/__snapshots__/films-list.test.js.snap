// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList is rendered correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/the-grand-budapest-hotel-poster.jpg"
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/0"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <video
        height="175"
        muted={true}
        poster="img/the-grand-budapest-hotel-poster.jpg"
        src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
</div>
`;
